src/MIDebugEngine/AD7.Impl/AD7Engine.cs

```diff
@@ -771,7 +771,19 @@ public int Detach()
         {
             _breakpointManager.ClearBoundBreakpoints();
 
-            _pollThread.RunOperation(() => _debuggedProcess.CmdDetach());
+            try
+            {
+                _pollThread.RunOperation(() => _debuggedProcess.CmdDetach());
+            }
+            catch (DebuggerDisposedException e)
+            {
+                // Detach command could cause DebuggerDisposedException and we ignore that.
+                if (e.AbortedCommand != "-target-detach")
```

Comment:
As of C# 6 you can use exception filters like so:

```csharp
catch (DebuggerDisposedException e) if (e.AbortedCommand != "-target-detach") 
{
    // Detach command could cause DebuggerDisposedException and we ignore that.
}
```

Does mono support this? More curious than a necessary change.